### DISCUSSION TITLE
No results from JS client calls to storage in local env edge-function

### DISCUSSION THREAD

**Discussion by joshmarnold** [supabase/supabase#13029] [Questions] [answered]
Hello, I'm working locally and am trying to get calls to the storage API in an edge-function to work. I've tried a couple of different calls like getBucket, listBuckets, etc.. and they all behave the same way where they return a 200 and no results. 

For example, I've tried to get a buckets details by doing:
const { data, error } = await supabaseClient.storage.getBucket("test");

This gives a status code 200 but no results. The bucket is there, has images in it and is public. I'm new to this so maybe it's policy related or config. related.

**Comment by GaryAustin1** [comment, marked as answer]
Also answered on your Discord.

You have to have and meet RLS policies for the storage.buckets table to access bucket calls.  You have to have storage.objects RLS policies to do file operations.   The only exception is for a public bucket you can get a public URL to read the file.  Nothing else changes for a public bucket versus private, all other requests must have and meet policies.

If you use service_role key you should bypass RLS though.

**Comment by joshmarnold** [comment]
Ohh, I gotcha. It was a policy issue where I didn't have any policies on the `storage.bucket` or `storage.object` tables 

1. I first confirmed this was a policy issue by using the `service_role_key` since it bypasses RLS which then returned items. Duh. It's all making sense now.

2. From there I just needed to add a more appropriate policy on `storage.buckets` like only allowing authenticated users to select.
